companion.py

```diff
@@ -105,51 +110,66 @@
 # In practice these arrays aren't very sparse so just convert them to lists with any 'gaps' holding None.
 def listify(thing):
     if thing is None:
-        return []	# data is not present
+        return []  # data is not present
+
     elif isinstance(thing, list):
-        return list(thing)	# array is not sparse
+        return list(thing)  # array is not sparse
+
     elif isinstance(thing, dict):
         retval = []
-        for k,v in thing.items():
+        for k, v in thing.items():
             idx = int(k)
+
             if idx >= len(retval):
                 retval.extend([None] * (idx - len(retval)))
                 retval.append(v)
             else:
                 retval[idx] = v
+
         return retval
+
     else:
-        assert False, thing	# we expect an array or a sparse array
-        return list(thing)	# hope for the best
+        assert False, thing  # we expect an array or a sparse array
```

Comment:
`assert False` always raises. Not entirely sure why this is here. Replacing it with a `raise ValueError` may be prudent

Reply:
Just be sure to check callers.  If they're catching exceptions including AssertionError then we need to be sure they'll catch ValueError.  Of course they're probably just doing bare `except:` ;) .

Reply:
Appears that the only caller is https://github.com/EDCD/EDMarketConnector/blob/develop/stats.py#L157 (according to my IDE and a `grep 'listify' **/*.py`, anyway). So probably okay to switch to a ValueError?

Reply:
Having traced that back up the possible call stacks, yeah, `raise ValueError` seems good.

Reply:
Raise added, with a message from the comment (and the given value, as the assert had)